lib/src/list.dart

```diff
@@ -85,15 +85,16 @@ class ManagedRealmList<T extends Object?> with RealmEntity, ListMixin<T> impleme
 
   @override
   bool remove(covariant T element) {
-    if (element is RealmObject && !element.isManaged) {
+    if (element is RealmObjectBase && !element.isManaged) {
```

Comment:
The same chack must be in `indexOf`. Replace `RealmObject` with `RealmObjectBase` in the folloowing code.
```dart
if (element is RealmObject && !element.isManaged) {
      throw RealmStateError('Cannot call indexOf on a managed list with an element that is an unmanaged object');
    }
```
I just can't add a comment on the right place.